packages/block-library/src/post-author/utils.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * WordPress dependencies
+ */
+import { createBlock } from '@wordpress/blocks';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Generate Author-related blocks based on block attributes.
+ *
+ * @param {Object} attributes Block's attributes.
+ *
+ * @return {Object} Generated block.
+ */
+export function migrateToRecommendedBlocks( attributes ) {
+	const {
+		avatarSize,
+		byline,
+		showBio,
+		isLink,
+		linkTarget,
+		textAlign,
+		style,
+		...restAttributes
+	} = attributes;
+
+	return createBlock(
+		'core/group',
+		{
+			...restAttributes,
+			style: {
+				...style,
+				color: {
+					...style?.color,
+					// Duotone must be applied to the avatar block.
+					duotone: undefined,
+				},
+			},
+			layout: {
+				type: 'flex',
+				flexWrap: 'nowrap',
+				verticalAlignment: 'top',
+			},
```

Comment:
The Post Author block also has a Flex layout, but seems to have [a 1em right margin](https://github.com/WordPress/gutenberg/blob/d339baf355b88e4b9a9c3434480824aa4d51f870/packages/block-library/src/post-author/style.scss#L12-L14) between the avatar and the content. Therefore, in a8173e3, this is used as the `styles.spacing.blockGap` value.

Reply:
Thanks for iterating on this. The block gap spacing is only one aspect of the issue. In the video I shared you'll notice that the overall width of the block change. That is unexpected and could further break themes.

#### Before

<img width="988" alt="Screenshot 2023-10-17 at 12 50 24 pm" src="https://github.com/WordPress/gutenberg/assets/60436221/3679b46f-6ab9-49fb-bd4e-9c80389eb5a0">

#### After

<img width="916" alt="Screenshot 2023-10-17 at 12 50 32 pm" src="https://github.com/WordPress/gutenberg/assets/60436221/a08fdb84-4567-49f0-95ad-8eeb92668fc1">

Reply:
Regarding the width of the block, I think the Post Author block does not have `box-sizing:border-box`, so the width change will occur if left and right padding is applied.

This spacing support for this block was added in #35963, but this PR was before #43243 that recommends adding `box-sizing:border-box` at the same time as supporting spacing controls.

Originally, I think the Post Author block should also have `box-sizing:border-box`, but what approach do you think is best to prevent breaking changes as much as possible? Personally, I think that increasing the width of the Post Author block by padding is not originally intended, and that changing the width may be acceptable for migration purposes.

Reply:
Update: Post Author block now supports borders, as of #64599. At the same time, `box-sizing: border-box` was added, so there should be no width change when migrating.